app/services/calculators/pension_contribution_calculator.rb

```diff
@@ -0,0 +1,53 @@
+module Calculators
+  class PensionContributionCalculator
+    Result = Data.define(:bank, :cash, :regular) do
+      def all_sources
+        bank + cash + regular
+      end
+
+      def self.blank
+        new(bank: 0, cash: 0, regular: 0)
+      end
+    end
+
+    class << self
+      def call(outgoings:, cash_transactions:, regular_transactions:, submission_date:, total_gross_income:)
+        if (outgoings + cash_transactions + regular_transactions).any?
+          monthly_bank = Calculators::MonthlyEquivalentCalculator.call(collection: outgoings)
+          monthly_cash = Calculators::MonthlyCashTransactionAmountCalculator.call(collection: cash_transactions)
+          monthly_regular = Calculators::MonthlyRegularTransactionAmountCalculator.result_for_transactions(regular_transactions)
+          monthly_source_total = monthly_bank + monthly_cash + monthly_regular
+          total = pension_contribution_cap(
+            submission_date:,
+            total_gross_income:,
+            monthly_pension_contribution: monthly_source_total,
+          )
+
+          pension_contribution = ->(monthly_source) { (monthly_source * (total / monthly_source_total)).round(2) }
```

Comment:
could this be simplified? I see the attraction, but it's a bit complicated compared with the simple duplicated version. e.g. could we just calculate the multiplicand (total / monthly_source_total) and then pass it to an ordinary method that calculates (a * b).round(2)?